app/models/user.rb

```diff
@@ -4,4 +4,36 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable,
          :confirmable, :lockable, :timeoutable, :trackable
+  devise :pwned_password unless Rails.env.test?
+
+  # Validations for username field
+  validates :username, presence:   true
+  validates :username, uniqueness: true
```

Comment:
Consider making username uniqueness validation case-insensitive using the `case_sensitive: false` option so variations of `denny` like `Denny` would be prevented.